sdk/cosmos/azure_data_cosmos/src/options/mod.rs

```diff
@@ -9,6 +9,19 @@ mod query_options;
 mod read_container_options;
 mod read_database_options;
 
+cfg_if::cfg_if! {
```

Comment:
Why can't you just use the built-in `cfg`? If it's a scoping thing, it only takes a couple extra lines:

```rust
#[cfg(feature = "control_plane")]
mod control_plane {
  mod create_container_options;
  // ...
  pub use create_container_options::CreateContainerOptions;
  // ...
}
#[cfg(feature = "control_plane")]
pub use control_plane::*;
```

Separately, why is there a feature called "control_plane" in a data plane SDK? If you need it, does it matter that you use a control plane API? Just wondering why it's optional.

Reply:
> Why can't you just use the built-in cfg? If it's a scoping thing, it only takes a couple extra lines

I can rework it to that.

> Separately, why is there a feature called "control_plane" in a data plane SDK?

Operations to create, update, and delete Databases and Containers are considered Control Plane operations, and they can be performed via ARM. However, the Cosmos DB **Data Plane** _also_ provides these APIs (which I've referred to here as "Control Plane Over Data Plane" operations). This SDK is calling those **Data Plane** operations. Hence the name of the feature flag being `control_plane`, though I appreciate it might not be the clearest name.

> Just wondering why it's optional.

The reason it's optional is something we raised earlier on in our discussions about this SDK. These "Control Plane Over Data Plane" operations _require_ key-based auth. They _cannot_ be performed with an Entra ID token. There is ongoing work to change this, but it's not committed to happening before this SDK GAs.

We _could_ just put these in the same `key_auth` feature flag as the rest of our key-based authentication, and now that I think through that, I might prefer that. I originally thought granularity might be good, but these APIs wouldn't be behind a feature flag _except_ for the fact that they depend on the `key_auth` feature, so there isn't really a need for a separate flag.

Reply:
I remembered that conversation, which is why I wondered why it was optional but, given `key_auth` also was (that conversation I clearly forgot), I agree it makes sense to unify them. I just didn't see much point in making those CP-over-DP features optional.